Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,425 @@
+					WeakReference<RadioButton> weakRef = elements[i];
+					if (weakRef.TryGetTarget(out RadioButton rb))
+					{
+						// Uncheck all checked RadioButtons different from the current one
+						if (rb != this && (rb.IsChecked == true) && rootScope == GetVisualRoot(rb))
+						{
+							rb.UncheckRadioButton();
+						}
+
+						++i;
+					}
+					else
+					{
+						// Remove dead instances
+						elements.RemoveAt(i);
+					}
+				}
+			}
+			else // Logical parent should be the group
+			{
+				Element parent = Parent;
+				if (parent != null)
+				{
+					// Traverse logical children
+					IEnumerable children = parent.LogicalChildren;
+					IEnumerator itor = children.GetEnumerator();
+					while (itor.MoveNext())
+					{
+						var rb = itor.Current as RadioButton;
+						if (rb != null && rb != this && string.IsNullOrEmpty(rb.GroupName) && (rb.IsChecked == true))
+						{
+							rb.UncheckRadioButton();
+						}
+					}
+				}
+			}
+		}
+
+		void UncheckRadioButton()
+		{
+			//IsChecked = false;
+			//SetValue(IsCheckedProperty, false);
+			SetValueCore(IsCheckedProperty, false, SetValueFlags.None, SetValuePrivateFlags.Silent);
+			OnPropertyChanged(nameof(IsChecked));
+		}
+
+		static void Register(RadioButton radioButton, string groupName)
+		{
+			if (_groupNameToElements == null)
+			{
+				_groupNameToElements = new Dictionary<string, List<WeakReference<RadioButton>>>(1);
+			}
+
+			if (_groupNameToElements.TryGetValue(groupName, out List<WeakReference<RadioButton>> elements))
+			{
+				// There were some elements there, remove dead ones
+				PurgeDead(elements, null);
+			}
+			else
+			{
+				elements = new List<WeakReference<RadioButton>>(1);
+				_groupNameToElements[groupName] = elements;
+			}
+
+			elements.Add(new WeakReference<RadioButton>(radioButton));
+		}
+
+		static void Unregister(RadioButton radioButton, string groupName)
+		{
+			if (_groupNameToElements == null)
+			{
+				return;
+			}
+
+			// Get all elements bound to this key and remove this element
+			if (_groupNameToElements.TryGetValue(groupName, out List<WeakReference<RadioButton>> elements))
+			{
+				PurgeDead(elements, radioButton);
+				if (elements.Count == 0)
+				{
+					_groupNameToElements.Remove(groupName);
+				}
+			}
+		}
+
+		static void PurgeDead(List<WeakReference<RadioButton>> elements, object elementToRemove)
+		{
+			for (int i = 0; i < elements.Count;)
+			{
+				WeakReference<RadioButton> weakRef = elements[i];
+				if (weakRef.TryGetTarget(out RadioButton rb))
+				{
+					++i;
+				}
+				else if (rb == elementToRemove)
+				{
+					elements.RemoveAt(i);
+				}
+			}
+		}
+
+		static Element GetVisualRoot(Element element)
+		{
+			Element parent = element.Parent;
+			while (parent != null && !(parent is Page))
+			{
+				parent = parent.Parent;
+			}
+			return parent;
+		}
+	}
+
+	public partial class RadioButton :
+		View,
+		IViewController,
+		IElementConfiguration<RadioButton>,
+		IButtonElement,
+		IButtonController,
```

Comment:
I don't think this should be a ButtonElement, and we shouldn't be implementing Pressed and Released. please discuss

Reply:
Pressed and Released are valid for VSM 

People might want to trigger some shadow or background effect when the user has touched the element